src/tests/ftest/container/BasicSnapshot.py

```diff
@@ -0,0 +1,215 @@
+sys.path.append('./util')
+sys.path.append('../util')
+sys.path.append('../../../utils/py')
+sys.path.append('./../../utils/py')
+
+# pylint: disable=import-error
+import ServerUtils
+import WriteHostFile
+from daos_api import DaosContext, DaosPool, DaosContainer, DaosSnapshot, DaosLog, DaosApiError
+
+# pylint: disable=broad-except
+class BasicSnapshot(Test):
+    """
+    DAOS-1370 Basic snapshot test
+
+    Test Class Description:
+    Test that a snapshot taken of a container remains unchaged even after an
+    object in the container has been updated 500 times.
+    Create the container.
+    Write an object to the container.
+    Take a snapshot.
+    Write 500 changes to the KV pair of the object.
+    Check that the snapshot is still there.
+    Confirm that the data in the snapshot is unchanged.
+    Destroy the snapshot
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(BasicSnapshot, self).__init__(*args, **kwargs)
+        self.snapshot = None
+
+    def setUp(self):
+
+        # get paths from the build_vars generated by build
+        with open('../../../.build_vars.json') as finput:
+            build_paths = json.load(finput)
+        basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+
+        server_group = self.params.get("server_group", '/server/',
+                                       'daos_server')
+
+        # setup the DAOS python API
+        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
+        self.d_log = DaosLog(self.context)
+
+        self.hostlist = self.params.get("test_machines", '/run/hosts/*')
+        hostfile = WriteHostFile.WriteHostFile(self.hostlist, self.workdir)
+
+        ServerUtils.runServer(hostfile, server_group, basepath)
+
+        # Set up the pool and container.
+        try:
+            # parameters used in pool create
+            createmode = self.params.get("mode", '/run/pool/createmode/')
+            createsetid = self.params.get("setname", '/run/pool/createset/')
+            createsize = self.params.get("size", '/run/pool/createsize/')
+            createuid = os.geteuid()
+            creategid = os.getegid()
+
+            # initialize a pool object then create the underlying
+            # daos storage
+            self.pool = DaosPool(self.context)
+            self.pool.create(createmode, createuid, creategid,
+                             createsize, createsetid, None)
+
+            # need a connection to create container
+            self.pool.connect(1 << 1)
+
+            # create a container
+            self.container = DaosContainer(self.context)
+            self.container.create(self.pool.handle)
+
+            # now open it
+            self.container.open()
+
+        except DaosApiError as error:
+            print(error)
+            print(traceback.format_exc())
+            self.fail("Test failed before snapshot taken")
+
+    def tearDown(self):
+        try:
+            self.container.close()
```

Comment:
Yo need to test `self.container` as being a valid object to ensure that `setUp()` got even as far as creating it and didn't error out before that.

```suggestion
            if self.container:
                self.container.close()
```

Reply:
Will do.